tests/dataset_runner.py

```diff
@@ -0,0 +1,180 @@
+import os
+import subprocess
+
+import requests
+import openpyxl
+
+from .data_store_agent import DataStoreAgent
+from .ingest_agents import IngestUIAgent, IngestApiAgent
+from .utils import Progress
+from .wait_for import WaitFor, TimedOut
+
+MINUTE = 60
+
+
+class DatasetRunner:
+
+    def __init__(self, deployment):
+        self.deployment = deployment
+        self.ingest_broker = IngestUIAgent(deployment=deployment)
+        self.ingest_api = IngestApiAgent(deployment=deployment)
+        self.data_store = DataStoreAgent(deployment=deployment)
+        self.dataset = None
+        self.submission_id = None
+        self.submission_envelope = None
+        self.upload_credentials = None
+        self.primary_to_results_bundles_map = {}
+        self.failure_reason = None
+
+    @property
+    def primary_bundle_uuids(self):
+        return list(self.primary_to_results_bundles_map.keys())
+
+    @property
+    def secondary_bundle_uuids(self):
+        return list(self.primary_to_results_bundles_map.values())
+
+    def run(self, dataset_fixture):
+        self.dataset = dataset_fixture
+        self.upload_spreadsheet_and_create_submission()
+        self.get_upload_area_credentials()
+        self.stage_data_files()
+        self.wait_for_envelope_to_be_validated()
+        self.complete_submission()
+        self.wait_for_primary_bundles_to_be_created()
+        self.wait_for_results_bundle_to_be_created()
+        if self.failure_reason:
+            raise RuntimeError(self.failure_reason)
+
+    def upload_spreadsheet_and_create_submission(self):
+        spreadsheet_filename = os.path.basename(self.dataset.metadata_spreadsheet_path)
```

Comment:
It would be helpful to change the project_shortname field in the spreadsheet each time the test is run, e.g. project_shortname="scale_test_2018_05_24.12_25_36". If we do that, we can set up green box so that it automatically labels all workflows with project_shortname, which is a first step towards being able to have the test query green box for status of test workflows. There is an [API](https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/update) for this. However, this could be added in a later PR.

Reply:
👍 later